File: services/go/spec/echo_service.go
package spec

import "errors"

type ServiceEcho struct{}

func (service *ServiceEcho) EchoBody(body *Message) (*EchoBodyResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *ServiceEcho) EchoQuery(intQuery int, stringQuery string) (*EchoQueryResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *ServiceEcho) EchoHeader(intHeader int, stringHeader string) (*EchoHeaderResponse, error) {
	return nil, errors.New("implementation has not added yet")
}
func (service *ServiceEcho) EchoUrlParams(intUrl int, stringUrl string) (*EchoUrlParamsResponse, error) {
	return nil, errors.New("implementation has not added yet")
}

File: services/ts-express-superstruct/services/echo_service.ts
import * as services from './spec/services'

export let echoService = (): services.EchoService => {
    let echoBody = async (params: services.EchoBodyParams): Promise<services.EchoBodyResponse> => {
        throw new Error('Not Implemented')
    }

    let echoQuery = async (params: services.EchoQueryParams): Promise<services.EchoQueryResponse> => {
        throw new Error('Not Implemented')
    }

    let echoHeader = async (params: services.EchoHeaderParams): Promise<services.EchoHeaderResponse> => {
        throw new Error('Not Implemented')
    }

    let echoUrlParams = async (params: services.EchoUrlParamsParams): Promise<services.EchoUrlParamsResponse> => {
        throw new Error('Not Implemented')
    }

    return {echoBody, echoQuery, echoHeader, echoUrlParams}
}
